junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertThrowsExactlyAssertionsTests.java

```diff
@@ -0,0 +1,324 @@
+		}
+		catch (AssertionError ex) {
+			assertMessageEquals(ex,
+				"Custom message ==> Expected java.io.IOException to be thrown, but nothing was thrown.");
+		}
+	}
+
+	@Test
+	void assertThrowsWithExecutableThatThrowsAnUnexpectedException() {
+		try {
+			assertThrowsExactly(IllegalStateException.class, (Executable) () -> {
+				throw new NumberFormatException();
+			});
+			expectAssertionFailedError();
+		}
+		catch (AssertionFailedError ex) {
+			assertMessageStartsWith(ex, "Unexpected exception type thrown ==> ");
+			assertMessageContains(ex, "expected: <java.lang.IllegalStateException>");
+			assertMessageContains(ex, "but was: <java.lang.NumberFormatException>");
+		}
+	}
+
+	@Test
+	void assertThrowsWithExecutableThatThrowsAnUnexpectedExceptionWithMessageString() {
+		try {
+			assertThrowsExactly(IllegalStateException.class, (Executable) () -> {
+				throw new NumberFormatException();
+			}, "Custom message");
+			expectAssertionFailedError();
+		}
+		catch (AssertionFailedError ex) {
+			// Should look something like this:
+			// Custom message ==> Unexpected exception type thrown ==> expected: <java.lang.IllegalStateException> but was: <java.lang.NumberFormatException>
+			assertMessageStartsWith(ex, "Custom message ==> ");
+			assertMessageContains(ex, "Unexpected exception type thrown ==> ");
+			assertMessageContains(ex, "expected: <java.lang.IllegalStateException>");
+			assertMessageContains(ex, "but was: <java.lang.NumberFormatException>");
+		}
+	}
+
+	@Test
+	void assertThrowsWithExecutableThatThrowsAnUnexpectedExceptionWithMessageSupplier() {
+		try {
+			assertThrowsExactly(IllegalStateException.class, (Executable) () -> {
+				throw new NumberFormatException();
+			}, () -> "Custom message");
+			expectAssertionFailedError();
+		}
+		catch (AssertionFailedError ex) {
+			// Should look something like this:
+			// Custom message ==> Unexpected exception type thrown ==> expected: <java.lang.IllegalStateException> but was: <java.lang.NumberFormatException>
+			assertMessageStartsWith(ex, "Custom message ==> ");
+			assertMessageContains(ex, "Unexpected exception type thrown ==> ");
+			assertMessageContains(ex, "expected: <java.lang.IllegalStateException>");
+			assertMessageContains(ex, "but was: <java.lang.NumberFormatException>");
+		}
+	}
+
+	@Test
+	@SuppressWarnings("serial")
+	void assertThrowsWithExecutableThatThrowsInstanceOfAnonymousInnerClassAsUnexpectedException() {
+		try {
+			assertThrowsExactly(IllegalStateException.class, (Executable) () -> {
+				throw new NumberFormatException() {
+				};
+			});
+			expectAssertionFailedError();
+		}
+		catch (AssertionFailedError ex) {
+			assertMessageStartsWith(ex, "Unexpected exception type thrown ==> ");
+			assertMessageContains(ex, "expected: <java.lang.IllegalStateException>");
+			// As of the time of this writing, the class name of the above anonymous inner
+			// class is org.junit.jupiter.api.AssertionsAssertThrowsTests$2; however, hard
```

Comment:
```suggestion
			// class is org.junit.jupiter.api.AssertThrowsExactlyAssertionsTests$2; however, hard
```

I know it's nitpicking, but... technically it's true. 😇